src/link.c

```diff
@@ -273,11 +275,12 @@ static LinkTable *single_LinkTable_new(const char *url)
     return linktbl;
 }
 
-LinkTable *LinkSystem_init(const char *url)
+LinkTable *LinkSystem_init(const char *f_url)
 {
     if (pthread_mutex_init(&link_lock, NULL)) {
         lprintf(error, "link_lock initialisation failed!\n");
     }
+    char *url = escape_full_url(f_url);
```

Comment:
What if individual files have weird filename?

Reply:
The existing codebase (before this PR) correctly handles individual filenames and directories. Consider the directory of my server already mounted in `foo` below. This also works in single file mode.

```
eyi@Zeero:~/Projects/HTTPDIRFS$ ls foo/
 appointment20231116-11-zyn4je.ics   Arduino   dipfree_en64.exe   dipfree_en.exe   happ:55  'SP ACE'   财富
eyi@Zeero:~/Projects/HTTPDIRFS$ ls foo/SP\ ACE/
'G B'  'SHA SUMS'
eyi@Zeero:~/Projects/HTTPDIRFS$ ls -l foo/财富
total 0
-r--r--r-- 1 eyi eyi 20 Apr 16 20:00 yam.txt
eyi@Zeero:~/Projects/HTTPDIRFS$ 
``` 

Our main problem is that we do not escape invalid characters in the base URL. As [this line](https://github.com/fangfufu/httpdirfs/blob/d6d4af0c8c4d2d505a53eba86f93ccd3109529f4/src/link.c#L581) shows, we just begin downloading the base URL without first escaping.

The whole essence of this PR (a wider scope than the previous hack of the space character) is to just escape the base URL.

The reason for the regex is because the server I was testing with was running locally on my PC on port 9000, since I could not find public servers with all sorts of funny URLs and files. 

So one of my test command was `httpdirfs -d -f --cache "http://0.0.0.0:9000/财富" foo`  

I could remove the regex part, but we still need to escape the base URL in addition to the files and directories at that location.

Reply:
No, what you suggested is unlikely the problem, because it works without the cache system enabled:
https://github.com/fangfufu/httpdirfs/issues/86#issuecomment-2045002052